### ISSUE TITLE
[feature] pouch cli supports search image

### ISSUE THREAD

**Issue by hellolijj** [alibaba/pouch#2839]
## Why you need it?

sometimes, i want to search some images  in terminal instead of web.

## How it could be?

```
[root@aliyun ~]# pouch search jvm
NAME                                              DESCRIPTION                                     STARS               OFFICIAL            AUTOMATED
clojure                                           Clojure is a dialect of Lisp that runs on th…   252                 [OK]
jruby                                             JRuby (http://www.jruby.org) is an implement…   83                  [OK]
travisci/ci-jvm                                                                                   4
heroku/jvm                                        Base Docker Image for JVM Languages on Heroku   2                                       [OK]
ofayau/busybox-jvm                                Prepare busybox to install a 32 bits JVM.       2                                       [OK]
gwilym/generic-jvm                                A generic container designed to run Java app…   1                                       [OK]
tdziurko/jvm-bloggers                             Docker image for https://github.com/tdziurko…   1
valentinomiazzo/jvm-memory-test                   A Docker image meant to experiment with the …   1                                       [OK]
cfje/jvmkill                                      jvmkill CI Image                                0
cfje/jvm-application-buildpack                                                                    0
codewars/jvm-runner-temp                                                                          0
codewars/jvm-runner                                                                               0
thomaschen/jvm_app                                                                                0
hustleinc/ci-jvm-base                             Image for building and deploying Hustle's JV…   0
hodaya/thousand-jvms-image                                                                        0
rawmind/alpine-jvm8                               This image is the jvm8 base. It comes from a…   0                                       [OK]
galacticfog/gestalt-laser-executor-jvm                                                            0
freckleiot/docker-jenkins-slave-jvm-build-tools   Jenkins slave image based on freckleiot/dock…   0                                       [OK]
uglyog/pact-jvm-server                            Standalone Pact JVM mock server                 0
pointhi/mx-jvmci-smt                              mx-jvmci including SMT solvers (z3, cvc4, bo…   0
jvmbloggers/jvm-bloggers                                                                          0
qualified/jvm-runner                                                                              0
projectsigma/jvm-base                                                                             0
sigmadocker/jvm-gcloud-base                                                                       0
jvmlimmat/jvm-aldryn-base-project                 JVM Aldryn Base Project                         0                                       [OK]
```
## Other related information

I find `api_images_search` is completed, but cli_image_search is absent.

**Comment by hellolijj** [comment]
today, I read some `imagesearch` code . i found this functions is no implemented. maybe i can do it.

**Comment by allencloud** [comment]
> today, I read some imagesearch code . i found this functions is no implemented. maybe i can do it.

That is so good to do this. While I am wondering if we could make it implement searching from user specified registry, like `pouch search -r a.b.rgistry.com nginx `. @hellolijj

**Comment by ZYecho** [comment]
fix by  #2850
